controllers/k8ssandra/schemas.go

```diff
@@ -117,7 +117,15 @@ func (r *K8ssandraClusterReconciler) checkInitialSystemReplication(
 		if err := json.Unmarshal([]byte(val), &replication); err == nil {
 			return replication, nil
 		} else {
-			return nil, err
+			// We could have an annotation in the old format. Try to parse it and convert it to the new format by pursuing the execution.
+			replicationOldFormat := make(map[string]interface{})
+			if err2 := json.Unmarshal([]byte(val), &replication); err2 == nil {
```

Comment:
Shouldn't we be unmarshalling into `replicationOldFormat` here?

The old schema was:

```go
type SystemReplication struct {
	Datacenters       []string `json:"datacenters"`
	ReplicationFactor int      `json:"replicationFactor"`
}
```
Wouldn't it be better to make sure we marshal into the expected types? 

Lastly, if the old format is detected, should we convert it to the new format?

Reply:
> Shouldn't we be unmarshalling into replicationOldFormat here?

Good catch!

> Wouldn't it be better to make sure we marshal into the expected types?

Sure thing 👍 

> Lastly, if the old format is detected, should we convert it to the new format?

I think that's the right thing to do indeed.